### PR TITLE
securityConfig mounts

### PR DESCRIPTION
**Pull request by bkomraz1**
### Description
Accurate mount of securityConfig files
 
### Issues Resolved
default files are disappear from securityConfig folder after mount of custom security file
[#207]
 
### Check List
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by peterzhuamazon** [comment]
@DandyDeveloper @TheAlgo please take a look and review this PR.
Thanks.

**Review by TheAlgo — COMMENTED** [review]
@bkomraz1 Can you take a look at the CI failures?

File: charts/opensearch/templates/statefulset.yaml
Comment (DandyDeveloper) [inline charts/opensearch/templates/statefulset.yaml:62]: There are other conditions where this basePath may be used. It'd be nice to have an `or` here to add in when the existing secret mount has been presented.

File: charts/opensearch/templates/statefulset.yaml
Comment (DandyDeveloper) [inline charts/opensearch/templates/statefulset.yaml:374]: Doesn't this need to be `name: security-config-{{ $key }} or some kind of unique identifier? Otherwise this breaks since you are using subPath and the secret can have multiple keys.

**Review by DandyDeveloper — CHANGES_REQUESTED** [review]
@bkomraz1 Have some concerns on the mounting, I could be wrong. I haven't testing this locally.

File: charts/opensearch/templates/statefulset.yaml
Comment (bkomraz1) [reply charts/opensearch/templates/statefulset.yaml:374]: You can use the same volume for mounting separate files (and not entirely volume content) by using subPath. And you can do it multiple times.

File: charts/opensearch/templates/statefulset.yaml
Comment (bkomraz1) [reply charts/opensearch/templates/statefulset.yaml:62]: It is only in the case when securityConfig.config.data is defined. This condition is from here:

```
{{- if .Values.securityConfig.config.data -}}
apiVersion: v1
kind: Secret
metadata:
  name: {{ include "opensearch.uname" . }}-securityconfig
```

So if chart is using exist secret, securityConfig.config.data can't be used. As result "securityconfig" secret will not be generated and securityconfigchecksum annotation will not be calculated.

File: charts/opensearch/templates/statefulset.yaml
Comment (DandyDeveloper) [reply charts/opensearch/templates/statefulset.yaml:374]: @bkomraz1 You can. Stupid me. Thank you.

**Review by DandyDeveloper — APPROVED** [review]

**Comment by peterzhuamazon** [comment]
@TheAlgo please approve and I will fix the changelogs and versions.
Thanks.

**Comment by bkomraz1** [comment]
@peterzhuamazon I resolved conflicts with latest main

**Comment by peterzhuamazon** [comment]
> @peterzhuamazon I resolved conflicts with latest main

Thanks, will wait for @TheAlgo approval on this.

**Comment by peterzhuamazon** [comment]
@TheAlgo can you give a approval if @bkomraz1 satisfied all the changes?
Thanks.

**Review by smlx — CHANGES_REQUESTED** [review]
This is a breaking change and would require a major version bump because it implements a change in behaviour: the security config you define is now _merged_ with the upstream demo security config, instead of replacing it.

IMO this PR implements surprising behaviour because with it any changes made to the demo security config in the upstream image will now impact my security config unless I have explicitly overridden it locally. The demo security config from the upstream image also has some stuff that should never make it into production e.g.
https://github.com/opensearch-project/security/blob/e0be1ebdfc8577fdbed7b636fc3bec03e90ca3a9/securityconfig/internal_users.yml#L11-L18

A better fix might be to either define a default set of security configs in `values.yaml` for `securityConfig.config.data."config.yml"` etc. Or alternatively to better document the way this currently behaves so that it doesn't surprise people.

**Comment by bkomraz1** [comment]
@smlx @peterzhuamazon 
I added support for securityConfig.config.dataComplete flag that is enabled by default. In this case usage of securityConfig.config.data will causes to disabling all default securityConfig files (it is current behaviour).
If value is set to false content of security folder will not disappear but specified files will be replaced.

**Comment by peterzhuamazon** [comment]
cc: @TheAlgo

**Comment by TheAlgo** [comment]
@DandyDeveloper @peterzhuamazon Do we want to bump the chart to `1.8.0` here?

**Comment by peterzhuamazon** [comment]
> @DandyDeveloper @peterzhuamazon Do we want to bump the chart to `1.8.0` here?

I will bump after you guys approve the code part.

**Review by TheAlgo — APPROVED** [review]
LGTM!. Safe to merge after minor chart bump

**Comment by peterzhuamazon** [comment]
Thanks @bkomraz1 for the contribution here!
It takes a long time to merge this.
Appreciate your help and support all the way.

**Pull request merged**
